### PR TITLE
Working on making the charts better

### PR DESCRIPTION
**Pull request by LindsayYoung**
Trying to get the totals to line up with what we had before.
- made changes suggested by Jon and Paul

### CODE REVIEW

**Comment by LindsayYoung** [comment]
Assumptions

 - fec_vsum_f76 is as amended
 - electioneering_com_vw is as amended
 - joining with things that match cmte_valid_fec_yr will get rid of out of cycle spenders

 Questions

 - paulclark.ref_pty_host_convention is used in the original script, How do I replicate that?

**Comment by LindsayYoung** [comment]
needs new test data

**Comment by LindsayYoung** [comment]
Paul - candidates
Jeff - parties
Carlo - PACs

**Comment by LindsayYoung** [comment]
most recent code comparison:
screen shot 2016-10-31 at 4 38 03 pm

**Comment by noahmanger** [comment]
Seeing in the above image that Candidates, PACs, and Party Committees are very close, I suggest that we push a release with just these three data sets. As I understand it, these are all really close, but occasionally slightly off due to the way that we're computing linkages. There will be two final tasks, which can happen later:

1. Add communication filers
2. Polish up linkages so that the main three are exact

@PaulClark2 : how does this sound to you? 

I've made a pull request for the front-end here https://github.com/18F/openFEC-web-app/pull/1673

If it's good, let's deploy these latest API and web app changes to the feature environment to verify totals and then if those are good we can do a mini deploy.

**Comment by PaulClark2** [comment]
That should be fine. I do have question about how far back we are going with these data in the API. We should only go back to 2007-2008. The formulas fall apart with data prior to 2007 -- mostly because the way we captured data, especially from paper reports, before October 2005. 

@LindsayYoung

**Comment by LindsayYoung** [comment]
Not a problem, @PaulClark2 I will add a filter to start that data at 2007.

Thank you @PaulClark2 @jwchumley @ccostino and @jontours for your help on this!

**Comment by PaulClark2** [comment]
Until we make a decision about host and convention committees, we should exclude these committees from the party totals.

C00578419	COMMITTEE ON ARRANGEMENTS FOR THE 2016 REPUBLICAN NATIONAL CONVENTION
C00485110	COMMITTEE ON ARRANGEMENTS FOR THE 2012 REPUBLICAN NATIONAL CONVENTION
C00422048	COMMITTEE ON ARRANGEMENTS FOR THE 2008 REPUBLICAN NATIONAL CONVENTION
	
C00567057	CLEVELAND 2016 HOST COMMITTEE INC
C00483586	2012 TAMPA BAY HOST COMMITTEE INC
C00431791	MINNEAPOLIS SAINT PAUL 2008 HOST COMMITTEE INC
	
C00571133	DEMOCRATIC NATIONAL CONVENTION COMMITTEE 2016
C00500405	2012 DEMOCRATIC NATIONAL CONVENTION COMMITTEE INC
C00435560	2008 DEMOCRATIC NATIONAL CONVENTION COMMITTEE, INC.
	
C00572958	PHILADELPHIA 2016 HOST COMMITTEE
C00493254 	COMMITTEE FOR CHARLOTTE/CHARLOTTE DNC HOST COMMITTEE
C00496570	NEW AMERICAN CITY INC
C00431593 	DENVER 2008 CONVENTION HOST COMMITTEE

**Comment by LindsayYoung** [comment]
This is the new result after removing host committees, as compared to the original calculations. 

screen shot 2016-11-02 at 11 40 54 am

File: data/sql_updates/large_aggregates.sql [hunk 1]
```diff
@@ -217,19 +246,41 @@ select
 from combined
 ;
 
--- creates cumulative table per cycle from the data receipts in the large aggregates
-drop table if exists entity_receipts_chart;
-create table entity_receipts_chart as (select idx, type, month, year, cycle, adjusted_total_receipts, sum(adjusted_total_receipts) OVER (PARTITION BY cycle, type order by year, month, type desc) from large_aggregates_tmp);
+-- creates cumulative materialized view per cycle from the data receipts in the large aggregates
+drop materialized view if exists entity_receipts_chart;
+create materialized view entity_receipts_chart as (
```
Comment (ccostino) [inline data/sql_updates/large_aggregates.sql]: Should this also be named with `_mv_tmp` at the end to ensure that it is renamed and refreshed properly?

File: data/sql_updates/large_aggregates.sql [hunk 2]
```diff
@@ -217,19 +246,41 @@ select
...
+);
+
+create unique index on large_aggregates_tmp (idx);
 
--- creates cumulative table per cycle from the data disbursements in the large aggregates
-drop table if exists entity_disbursements_chart;
-create table entity_disbursements_chart as (select idx, type, month, year, cycle, adjusted_total_disbursements, sum(adjusted_total_disbursements) OVER (PARTITION BY cycle, type order by year, month, type desc) from large_aggregates_tmp);
+-- creates cumulative materialized view per cycle from the data disbursements in the large aggregates
+drop materialized view if exists entity_disbursements_chart;
+create materialized view entity_disbursements_chart as (
```
Comment (ccostino) [inline data/sql_updates/large_aggregates.sql]: Same note as above with this view name.

File: data/sql_updates/large_aggregates.sql [hunk 3]
```diff
@@ -217,19 +246,41 @@ select
...
+        year,
+        cycle,
+        adjusted_total_disbursements,
+        sum(adjusted_total_disbursements) OVER (PARTITION BY cycle, type order by year, month, type desc)
+    from large_aggregates_tmp
+    where cycle >= 2008
+);
 
--- don't need this after making the charts
-drop table if exists large_aggregates_tmp;
```
Comment (ccostino) [inline data/sql_updates/large_aggregates.sql]: We don't need to drop this anymore once done?

File: data/sql_updates/omnibus_dates.sql [hunk 4]
```diff
@@ -64,7 +64,7 @@ with elections_raw as(
                 ], '-')
             else elections_raw.election_state
         end as report_contest
-    from disclosure.trc_report_due_date reports
+    from trc_report_due_date reports
```
Comment (ccostino) [inline data/sql_updates/omnibus_dates.sql]: Did this change get rolled in due to switching the target branch from `develop` to `master`?  Will this be okay?

Comment (LindsayYoung) [inline data/sql_updates/omnibus_dates.sql]: Great catch!

File: data/sql_updates/large_aggregates.sql (hunk 3 above)
Comment (LindsayYoung) [inline data/sql_updates/large_aggregates.sql]: I think it has to exist since we have additional views on top of it

**Comment by ccostino** [comment]
Thank you, @LindsayYoung, @PaulClark2, @jwchumley, and @jontours!

**Pull request merged**